### PR TITLE
add version number

### PR DESCRIPTION
**Pull request by JamesRobertsonGames**

### CODE REVIEW

**Comment by sumit0190** [comment]
There's two things we need to be careful of here:
1. We should use the `igraph` that we've built recently, instead of using the vendored one. There's a [section](https://github.com/igraph/python-igraph/#linking-to-an-existing-igraph-installation) in the README that explains this.
2. The external `igraph` would only work with Linux and macOS, since Windows is not supported right now (the link explains it). This means we'd have to compile the vendored version, which should work, but it probably doesn't right now because you are trying to compile it statically and it needs to resolve all references at link time.
``` 
diff --git a/recipe/bld.bat b/recipe/bld.bat
index fac7e93..c21e4f0 100644
--- a/recipe/bld.bat
+++ b/recipe/bld.bat
@@ -12,14 +12,12 @@ set IGRAPH_CMAKE_EXTRA_ARGS=%CMAKE_ARGS%^
       -DIGRAPH_USE_INTERNAL_LAPACK=0 ^
       -DIGRAPH_USE_INTERNAL_ARPACK=1 ^
       -DIGRAPH_USE_INTERNAL_GLPK=0 ^
-      -DIGRAPH_USE_INTERNAL_CXSPARSE=0 ^
       -DIGRAPH_USE_INTERNAL_GMP=1 ^
-      -DBUILD_SHARED_LIBS=off ^
+      -DBUILD_SHARED_LIBS=on ^
       -DIGRAPH_ENABLE_LTO=1 ^
       -DIGRAPH_ENABLE_TLS=1 ^
       -DIGRAPH_GRAPHML_SUPPORT=1
-set IGRAPH_EXTRA_LIBRARIES=cxsparse,glpk,libxml2
-
+set IGRAPH_EXTRA_LIBRARIES=glpk,libxml2
 
 %PYTHON% -m pip install --no-deps --ignore-installed --no-build-isolation . -vvv
 if errorlevel 1 exit 1
diff --git a/recipe/meta.yaml b/recipe/meta.yaml
index f7fe96d..5635d25 100644
--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -32,12 +32,8 @@ requirements:
     - {{ compiler('cxx') }}
     - cmake
     - ninja
-    - llvm-openmp  # [osx]
     - make  # [not win]
     - pkg-config  # [not win]
-    - intel-openmp 2023.1.0  # [win]
-    - mkl-devel 2023.1.0  # [win]
-    - mkl 2023.1.0  # [win]
   host:
     - glpk 4.65  # [not (linux and s390x)]
     - libxml2  {{ libxml2 }}
@@ -45,6 +41,10 @@ requirements:
     - gmp 6.2.1  # [not win]
     - mpir 3.0.0  # [win]
     - openblas {{ openblas }}  # [not win]
+    - intel-openmp 2023.1.0  # [win]
+    - llvm-openmp  # [osx]
+    - mkl-devel 2023.1.0  # [win]
+    - mkl 2023.1.0  # [win]
     - python
     - pip
     - wheel
```
This should work, I think.

**Comment by akabanovs** [comment]
To reviewers of this PR! 
I am not sure if mpir (Multiple Precision Integers and Rationals) is required for win, used in CF recipe. Mpir package seems to contain some GMP stuff (GNU Multiple Precision Arithmetic Library). However, we are using internal GMP on windows...

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,69 +10,75 @@ source:
...
+  patches:
+    - 0001-add-igraph-libs.patch  # [win]
 
 build:
   number: 0
   skip: true  # [win and vc<14]
-  script_env:
-    - F2C_EXTERNAL_ARITH_HEADER={{ RECIPE_DIR }}/arith_arm64.h  # [arm64]
-
+  skip: true  # [linux and ppc64le]
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:19]: Could you write a comment on why we skip `ppc64le`?

**Review by skupr-anaconda — APPROVED** [review]
LGTM

File: recipe/meta.yaml (hunk 1 above)
Comment (akabanovs) [reply recipe/meta.yaml:19]: Will do. It is because we are not building arpack on linux-ppc64le, which is required by igraph and eventually python-igraph.

**Review by sumit0190 — APPROVED** [review]
We might want to investigate if some of the libraries are being statically linked and therefore not needed at runtime (like GLPK), but in the interest of time we should probably do that later; this is great for now.
(Same for MPIR btw, it's not needed but let's leave it for now).

**Pull request merged**